app/components/ProductTile.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { getProductThunk } from '../redux/singleProduct.js';
+import { makeStyles } from '@material-ui/core/styles';
+import { Link } from 'react-router-dom';
+import Card from '@material-ui/core/Card';
+import CardActionArea from '@material-ui/core/CardActionArea';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+
+const ProductTile = props => {
+  useEffect(() => {
+    const { id } = props.match.params;
+    props.getProduct(id);
+  }, []);
+
+  const product = props.product;
+  const useStyles = makeStyles({
+    card: {
+      maxWidth: 200
+    },
+    media: {
+      height: '100%'
+    }
+  });
+  const classes = useStyles();
+
+  if (!product.name) {
+    return <div>Product not found...</div>;
+  } else {
+    return (
+      <div>
+        <Card className={classes.card}>
+          <Link to={`/products/${product.id}`}>
+            <CardActionArea>
+              <CardMedia
+                component="img"
+                height="140"
+                image={product.imageUrl}
+              />
+            </CardActionArea>
+          </Link>
+          <CardActions>
+            <Button
+              size="small"
+              color="primary"
+              onClick={() => props /*props.singleProduct(product)}*/}
```

Comment:
I would move this to the link because that way a single product will be selected when the image is clicked as well